control-plane/connect-inject/controllers/pod/pod_controller.go

```diff
@@ -145,10 +145,22 @@ func (r *Controller) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	if hasBeenInjected(pod) {
 		if err := r.writeProxyConfiguration(ctx, pod); err != nil {
+			// We could be racing with the namespace controller.
+			// Requeue (which includes backoff) to try again.
+			if common.ConsulNamespaceIsNotFound(err) {
+				r.Log.Info("Consul namespace not found; re-queueing request",
+					"pod", req.Name, "ns", req.Namespace, "consul-ns", r.getConsulNamespace(req.Namespace))
+				return ctrl.Result{Requeue: true}, nil
+			}
 			errs = multierror.Append(errs, err)
 		}
 
 		if err := r.writeWorkload(ctx, pod); err != nil {
+			if common.ConsulNamespaceIsNotFound(err) {
+				r.Log.Info("Consul namespace not found; re-queueing request",
+					"pod", req.Name, "ns", req.Namespace, "consul-ns", r.getConsulNamespace(req.Namespace))
+				return ctrl.Result{Requeue: true}, nil
+			}
```

Comment:
In theory is the previous write succeeds, this is unnecessary, but I kept it in case it gets refactored later.

Reply:
Should we add a comment to that effect? That way someone doesn't come through here and "optimize" not realising your intent.